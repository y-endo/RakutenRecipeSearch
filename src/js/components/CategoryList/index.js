import React from 'react';
import PropTypes from 'prop-types';

import CategoryItem from '@/js/components/CategoryItem/index';

export default class CategoryList extends React.Component {
  render() {
    const listItems = this.props.recipeCategoryListData.map((data, index) => {
      return <CategoryItem key={index} recipeCategoryData={data} />;
    });

    return <ul className="recipe-list">{listItems}</ul>;
  }
}

CategoryList.propTypes = {
  recipeCategoryListData: PropTypes.array
};
