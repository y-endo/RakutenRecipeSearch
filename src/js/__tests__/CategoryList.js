import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CategoryList from '@/js/components/CategoryList/index';

test('カテゴリーリストが正しく表示されるか', () => {
  const testData = [
    {
      categoryId: 10,
      categoryName: 'カテゴリー1'
    },
    {
      categoryId: 20,
      categoryName: 'カテゴリー2'
    },
    {
      categoryId: 30,
      categoryName: 'カテゴリー3'
    },
    {
      categoryId: 40,
      categoryName: 'カテゴリー4'
    }
  ];
  const wrapper = shallow(<CategoryList recipeCategoryListData={testData} />);

  // カテゴリーアイテムの数が正しい
  expect(wrapper.find('CategoryItem')).toHaveLength(4);

  // スナップショット
  expect(toJson(wrapper)).toMatchSnapshot();
});
