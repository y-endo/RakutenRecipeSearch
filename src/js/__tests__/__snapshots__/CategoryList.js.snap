// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`カテゴリーリストが正しく表示されるか 1`] = `
<ul
  className="recipe-list"
>
  <CategoryItem
    key="0"
    recipeCategoryData={
      Object {
        "categoryId": 10,
        "categoryName": "カテゴリー1",
      }
    }
  />
  <CategoryItem
    key="1"
    recipeCategoryData={
      Object {
        "categoryId": 20,
        "categoryName": "カテゴリー2",
      }
    }
  />
  <CategoryItem
    key="2"
    recipeCategoryData={
      Object {
        "categoryId": 30,
        "categoryName": "カテゴリー3",
      }
    }
  />
  <CategoryItem
    key="3"
    recipeCategoryData={
      Object {
        "categoryId": 40,
        "categoryName": "カテゴリー4",
      }
    }
  />
</ul>
`;
